### PR TITLE
chore: add support for Angular 16

### LINKED ISSUE #26044: Add Component Testing Support for Angular 16
### What would you like?

With the release of [Angular 16](https://www.npmjs.com/package/@angular/core/v/16.0.0-next.1) just around the corner, we need to validate that it does not break Angular Component Testing.
- Angular 16 adds support for `signals` so we need to validate it works as expected and provide examples in our system tests for examples with signals.
- Validate the builder we are using doesn't have any breaking changes

### Why is this needed?

We want to make sure we are providing CT support for Angular users as close to release of 16 as possible.

### Other

_No response_

### PR DESCRIPTION
**Pull request by jordanpowell88**
- Closes #26044 

### Additional details
This PR adds support for Angular 16 (set to release in the coming weeks). 
- also adds tests to validate `signal`(s) in Angular will work in CT
- moves the `standalone.component.ts` tests that were included in `system-tests/project` angular versions `14, 15, and now 16` into `project-fixtures/angular`.

### Steps to test
1. change `ANGULAR_MAJOR_VERSIONS` in `system-tests/tests/component_testing_spec.ts` to just test version `16`
2. Run `yarn workspace @tooling/system-tests test component_testing_spec.ts`

### How has the user experience changed?
The user experience should remain the same

### PR Tasks

- [X] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by jordanpowell88** [comment]
https://github.com/angular/angular/pull/46641

**Comment by jordanpowell88** [comment]
We need to add support for `output()` for `createSpy` and `autoOutputSpy`

**Comment by jordanpowell88** [comment]
https://github.com/angular/angular-cli/pull/24923

**Comment by jordanpowell88** [comment]
We will need to create a follow up issue to validate/add support for `esbuild` support that ships with Angular 16

File: system-tests/test/component_testing_spec.ts [hunk 1]
```diff
@@ -127,7 +127,7 @@ describe(`Angular CLI major versions`, () => {
       spec = `${spec},!src/app/components/standalone.component.cy.ts`
     }
 
-    systemTests.it(`v${majorVersion} with mount tests`, {
+    systemTests.it.only(`v${majorVersion} with mount tests`, {
```
Comment (mike-plummer) [inline system-tests/test/component_testing_spec.ts]: Intentional `.only`?

**Comment by astone123** [comment]
@jordanpowell88 should we add the `angular-16` project to the webpack-dev-server e2e tests here (probably could also update that list to be `WEBPACK_ANGULAR` instead, looks like copy pasta)? https://github.com/cypress-io/cypress/blob/baaf717823d13c31a05c2c68930d80b9ae1fd847/npm/webpack-dev-server/cypress/e2e/angular.cy.ts#L5

**Review by lmiller1990 — APPROVED** [review]

**Comment by lmiller1990** [comment]
CI green! Just need one more ✅ , thanks @astone123 for the drive by fixes.

**Review by mike-plummer — APPROVED** [review]

**Pull request merged**